### ISSUE TITLE
Bot Connector Reply to Activity API returning ambiguous HTTP 400

### ISSUE THREAD

**Issue by peterinnesmsft** [microsoft/BotFramework-Services#105]
My team is encountering an issue when attempting to send a reply to users in Microsoft Teams containing a large Adaptive Card attachment via the Reply to Activity API. We are receiving an HTTP 400 response with the following fault returned:

{"error":{"code":"ServiceError","message":"Unknown"}}

I am completely unable to reasonably determine based on this fault code what exactly I am doing incorrectly. Through experimentation, I seem to have been able to place it down to an overly large request payload being sent to the Connector service - however, I cannot be 100% certain that is the issue.

I have attempted a few troubleshooting steps, such as verifying that my request payload is not being truncated when sent via HTTP, ensuring the Adaptive Card itself is valid via testing the in the Adaptive Cards Designer as well as the Microsoft Teams Card Builder, and everything would suggest that the JSON payload being sent should be valid.

What makes matters more confusing is that, if this truly is an issue caused by an overly large payload, the same payload of equivalent size was working perfectly fine little over two weeks ago, but seems to have mysteriously stopped working without any changes to the code to construct our replies or the card attachment itself. I've also verified that the backing data sourced from external sources is valid as expected.

I can provide additional data offline to help with any investigation, but ultimately I have two issues I need assistance with:

1. Help me understand what is causing the HTTP 400 so that we can address any issues on our side.
2. Improve the HTTP Fault Codes returned to be more descriptive and enable developers to better self-diagnose issues in cases such as this.

Please let me know if there's any additional information that I can provide in the meantime.

Thanks,
Peter

**Comment by jameslew** [comment]
+ some folks with deep knowledge on MSTeams.

**Comment by weskroesbergen** [comment]
We're running into this as well, which *appears* also related to payload size. It would help if there was a documented payload size at the very least to determine if that is indeed the case. There's nothing useful logged on the service-side channel either. The generated JSON runs fine in AC designer, and no issues reported in linters either.

**Comment by jameslew** [comment]
Thanks @weskroesbergen and @peterinnesmsft .  I've been working with the MS Teams group on this, and it turns out that this is a hard-coded size limit that has recently become more strictly enforced.  They are both working on documentation for this as well as looking at what can be done to increase this limit.  I do not have an ETA yet, only that it's being investigated.

**Comment by jameslew** [comment]
I can confirm that this has been added to the Microsoft Teams backlog, but do not yet have an ETA, and assume it would go out as part of a future feature release rather than a short term bug fix.  In the interim, they've suggested using Task Modules as a workaround:

Our recommendation for now is to use a task-module to show the same Adaptive Card in a pop-over or a web-view (if you wish to have more complex type-ahead matching for example). Documentation on using Task Modules with Bots is [here](https://docs.microsoft.com/en-us/microsoftteams/platform/concepts/task-modules/task-modules-bots).

Thank you!

**Comment by siddharthpant92** [comment]
Hi,
I came across this thread when I tried to find a solution to my problem. I'm getting the same error as mentioned originally (ServiceError). I've just started building a bot and was trying to follow the steps mentioned here https://docs.microsoft.com/en-us/azure/bot-service/rest-api/bot-framework-rest-connector-send-and-receive-messages?view=azure-bot-service-4.0#send-message . But when i send a message to the endpoint `v3/conversations/{conversationId}/activities`, the response I get is 
```
{
    "error": {
        "code": "ServiceError",
        "message": "Unknown"
    }
}
```
I saw that this issue was marked Closed and not resolved or anything to suggest that a solution was made available. 
Just for clarity, i am specifically trying to implement the feature where the bot can start a conversation with a user with a simple text message, for which we do not want send a task-module thats shows an adaptive card
